### PR TITLE
[AU4] Added some built-in effects. Step 2

### PR DESCRIPTION
**Pull request by igorkorsukov**
I don't know yet how it will all work, set parameters, apply...

This is only:
* Defining the file structure for built-in effects
* Registering effects and view models
* View blanks
* Open effect dialog (was already done, just slightly corrected)

### CODE REVIEW

**Review by Eism — APPROVED** [review]

File: au4/src/effects/CMakeLists.txt [hunk 1]
```diff
@@ -30,6 +30,11 @@ set(MODULE_SRC
     ${CMAKE_CURRENT_LIST_DIR}/internal/effectsuiengine.cpp
     ${CMAKE_CURRENT_LIST_DIR}/internal/effectsuiengine.h
 
+    # internal/au3
+    ${CMAKE_CURRENT_LIST_DIR}/internal/au3/buildineffects.cpp
```
Comment (saintmatthieu) [inline au4/src/effects/CMakeLists.txt]: builtin (with t)

File: au4/src/effects/effectsmodule.cpp [hunk 2]
```diff
@@ -86,20 +87,21 @@ void EffectsModule::registerUiTypes()
...
 }
 
-void EffectsModule::onInit(const muse::IApplication::RunMode& mode)
+void EffectsModule::onInit(const muse::IApplication::RunMode&)
 {
-    if (mode != muse::IApplication::RunMode::GuiApp) {
-        return;
-    }
+    //! NOTE Should be before PluginManager::Get().Initialize
+    BuildInEffects::init();
```
Comment (saintmatthieu) [inline au4/src/effects/effectsmodule.cpp:93]: Here too, with t.

File: au4/src/effects/internal/au3/buildineffects.cpp [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+{
+    static BuiltinEffectsModule::Registration< au::effects::EffectAmplify > regAmplify;
+    static BuiltinEffectsModule::Registration< au::effects::EffectChirp > regChirp;
+    static BuiltinEffectsModule::Registration< au::effects::EffectTone > regTone;
+}
+
+EffectMetaList BuildInEffects::effectMetaList() const
+{
+    EffectMetaList list;
+    auto range = PluginManager::Get().PluginsOfType(PluginTypeEffect);
```
Comment (saintmatthieu) [inline au4/src/effects/internal/au3/buildineffects.cpp]: Why don't you guys ever seem to use `const` ?

File: au4/src/effects/effectstypes.h [hunk 4]
```diff
@@ -32,4 +32,7 @@ struct EffectCategory {
 };
 
 using EffectCategoryList = std::vector<EffectCategory>;
+
+constexpr const char16_t* VST_CATEGORY_ID = u"vst";
+constexpr const char16_t* BUILDIN_CATEGORY_ID = u"buildin";
```
Comment (saintmatthieu) [inline au4/src/effects/effectstypes.h:37]: t

File: au4/src/effects/view/effectbuilder.cpp [hunk 5]
```diff
@@ -47,9 +49,9 @@ void EffectBuilder::load(const QString& id, QObject* itemParent)
 
     //! NOTE We create extension UI using a separate engine to control what we provide,
     //! making it easier to maintain backward compatibility and stability.
-    QQmlComponent component = QQmlComponent(engin, meta.url.toQString());
+    QQmlComponent component = QQmlComponent(engin, meta.qmlUrl.toQString());
```
Comment (saintmatthieu) [inline au4/src/effects/view/effectbuilder.cpp:52]: "engine" ?

**Review by saintmatthieu — APPROVED** [review]
Excellent !
Looking forward to those QML components coming to life.
Please address the "buildin" spelling mistake, hurts a little when reading :)

**Pull request merged**